buildpack.md

```diff
@@ -306,36 +312,37 @@ At the end of each individual buildpack's build phase:
 
 ### Purpose
 
-The purpose of detection is to find an ordered group of buildpacks to use during the build phase.
-These buildpacks must be compatible with the app.
+The purpose of detection is to find an ordered group of image extensions and executable buildpacks to use during the generation and build phases.
```

Comment:
I went with the terminology proposed [here](https://github.com/buildpacks/spec/pull/300/files). I can update this PR with feedback from that PR.

Reply:
TODO: update to change "executable buildpacks" -> "component buildpacks"